src/multiint.cpp

```diff
@@ -2459,6 +2495,157 @@ static void loadMapSettings2()
+		}
+		if (!locked.bases)
+		{
+			game.base = rand() % 3;
+			((MultichoiceWidget *)widgGetFromID(psWScreen, MULTIOP_BASETYPE))->choose(game.base);
+		}
+
+		// Structure limits are simply 0 or max, only for NO_TANK, NO_CYBORG, NO_VTOL, NO_UPLINK, NO_LASSAT.
+		if (!bLimiterLoaded || !asStructureStats)
+		{
+			initLoadingScreen(true);
+			if (resLoad("wrf/limiter_data.wrf", 503))
+			{
+				bLimiterLoaded = true;
+			}
+			closeLoadingScreen();
+		}
+		for (int i = 0; i < ARRAY_SIZE(limitIcons) - 1; ++i)	// skip last item, MPFLAGS_FORCELIMITS
+		{
+			randomizeLimit(limitIcons[i].stat);
+		}
+		if (rand() % 2 == 0)
+		{
+			ingame.flags |= MPFLAGS_FORCELIMITS;
+		}
+		else
+		{
+			ingame.flags &= ~MPFLAGS_FORCELIMITS;
+		}
+		createLimitSet();
+		applyLimitSet();
+		updateLimitFlags();
+		updateLimitIcons();
+
+		// Pick a map for a number of players and tech level
+		game.techLevel = current_tech = (rand() % 4) + 1;
+		LEVEL_LIST levels;
+		do
+		{
+			// don't kick out already joined players because of randomize
+			int players = NET_numHumanPlayers();
+			int minimumPlayers = std::max(players, 2);
+			current_numplayers = minimumPlayers;
+			if (minimumPlayers < MAX_PLAYERS_IN_GUI)
+			{
+				current_numplayers += (rand() % (MAX_PLAYERS_IN_GUI - minimumPlayers));
+			}
+			levels = enumerateMultiMaps(game.techLevel, current_numplayers);
+		}
+		while (levels.empty()); // restart when there are no maps for a random number of players
+
+		int pickedLevel = rand() % levels.size();
+		LEVEL_DATASET *mapData = nullptr;
+		for (LEVEL_LIST::iterator it = levels.begin(); it != levels.end(); ++it, --pickedLevel)
+		{
+			if (pickedLevel <= 0)
+			{
+				mapData = *it;
+				break;
+			}
+		}
+		loadMapSettings0(mapData);
+		loadMapSettings1();
+		loadMapPreview(false);
+
+		// Reset and randomize player positions, also to guard
+		// against case where in the previous map some players
+		// took slots, which aren't available anymore
+
+		// First, put human players at the top
+		int pos = 0;
+		for (int i = 0; i < MAX_PLAYERS; i++)
+		{
+			if (isHumanPlayer(i))
+			{
+				NetPlay.players[i].position = pos;
+				NetPlay.players[i].team = pos;
+				NETBroadcastPlayerInfo(i);
+				pos++;
+			}
+		}
+
+		// Fill with AIs if places remain
+		for (int i = 0; i < current_numplayers; i++)
+		{
+			if (!isHumanPlayer(i))
+			{
+				NetPlay.players[i].position = pos;
+				NetPlay.players[i].team = pos;
+				NETBroadcastPlayerInfo(i);
+				pos++;
+			}
+		}
+		// Reset remaining positions
+		for (int i = current_numplayers; i < MAX_PLAYERS; i++)
+		{
+			NetPlay.players[i].allocated = false;
+			NETBroadcastPlayerInfo(i);
+		}
+		// Randomize positions
+		for (int i = 0; i < current_numplayers; i++)
+		{
+			SendPositionRequest(i, NetPlay.players[rand() % current_numplayers].position);
+		}
+		netPlayersUpdated = true;
+
+		if (bMultiPlayer)
+		{
+			resetReadyStatus(true);
+		}
+
+		if (bHosted)
+		{
+			disableMultiButs();
+		}
```

Comment:
Setting `netPlayersUpdated = true` is unnecessary.

I'd suggest:

```
if (bHosted)
{
	resetReadyStatus(true);
	disableMultiButs();
}
```

Which should stop that host-only related info spam.

Reply:
OK, thanks!